### ISSUE TITLE
Add a little more detail to "components"

### ISSUE THREAD

**Issue by matozoid** [javaparser/javaparser-website#17]
It would be good to mention the following "components" or "features" with a one or two line description:
- pretty printing
- meta model
- CST
- lexical preservation
- mutable AST
- visitors
- tokens attached to nodes
- comment assignment algorithm
- Javadoc parser/printer
- Source root

**Comment by ftomassetti** [comment]
True. Some of them are user facing features, while others (CST, meta model) are more implementation information relevant to contributors or very advanced users

**Comment by matozoid** [comment]
The CST certainly pretty prints well, though.

**Comment by ftomassetti** [comment]
Should we have a specific page named components? Should we list those in the homepage?

**Comment by matozoid** [comment]
I've used "features" instead of "components" in our secret document now, that seems to cover the topic better. I propose something like "Symbol Solver - (description of what it does) - (description of what to expect when using it)"

**Comment by matozoid** [comment]
Oh, yeah, we can make a separate page or not. You're in charge of the site :-)
